src/components/login/Login.js

```diff
@@ -1,23 +1,43 @@
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 import { withRouter, Route } from 'react-router-dom'
+import { parse } from 'query-string'
 
 import LoginContainer from './LoginContainer'
 import LoginForm from './LoginForm'
 import LoginConfirm from './LoginConfirm'
 import LoginDetails from './LoginDetails'
+import LoginIncorrectContractId from './LoginIncorrectContractId'
 
-import { handleRefreshAccount, handleRefreshUrl, switchAccount, clearAlert, allowLogin, handleLoginUrl } from '../../actions/account'
+import { handleRefreshAccount, handleRefreshUrl, switchAccount, clearAlert, allowLogin, handleLoginUrl, checkContract } from '../../actions/account'
 
 class Login extends Component {
     state = {
         buttonLoader: false,
         dropdown: false,
+        contractCorrect: false
     }
 
     componentDidMount = () => {
         this.props.handleLoginUrl()
-    }    
+
+        const { checkContract, history, match, account: { url } } = this.props
+        const contractId = url.contract_id || parse(history.location.search).contract_id
+
+        if (contractId && !this.state.contractCorrect) {
+            checkContract(contractId)
+                .then(({ error }) => {
+                    if (error) {
+                        history.push(`${match.path}/incorrect-contract-id`)
+                        return
+                    }
+
+                    this.setState(() => ({
```

Comment:
This kinda defeats whole purpose of using Redux. If state isn't updated by reducer – there isn't much sense in using Redux.